spec/lib/saml/saml_settings_service_spec.rb

```diff
@@ -1,36 +1,46 @@
 # frozen_string_literal: true
 require 'rails_helper'
 require 'saml/settings_service'
+require 'lib/sentry_logging_spec_helper'
 
 RSpec.describe SAML::SettingsService do
+  let(:success_response_body) { File.read("#{::Rails.root}/spec/fixtures/files/saml_xml/metadata_response_body.txt") }
   describe '.saml_settings' do
     context 'with a 200 response' do
+      before { stub_request(:get, Settings.saml.metadata_url).to_return(status: 200, body: success_response_body) }
       it 'should only ever make 1 external web call' do
-        stub_request(:get, Settings.saml.metadata_url).to_return(status: 200, body: 'abc')
-        SAML::SettingsService.saml_settings
+        SAML::SettingsService.merged_saml_settings(true)
         SAML::SettingsService.saml_settings
         SAML::SettingsService.saml_settings
         expect(a_request(:get, Settings.saml.metadata_url)).to have_been_made.at_most_once
       end
       it 'returns a settings instance' do
-        expect(SAML::SettingsService.saml_settings(true)).to be_an_instance_of(OneLogin::RubySaml::Settings)
+        expect(SAML::SettingsService.merged_saml_settings(true)).to be_an_instance_of(OneLogin::RubySaml::Settings)
       end
     end
     context 'with metadata 500 responses' do
-      it 'should log three attempts' do
+      before do
         stub_request(:get, Settings.saml.metadata_url).to_return(
           status: 500, body: 'bad news bears'
         )
+      end
+      it 'should log three attempts' do
         expect(Rails.logger).to receive(:warn).exactly(2).times.with(/Failed to load SAML metadata: 500: try \d of 3/)
         expect(Rails.logger).to receive(:error).exactly(1).times.with(/Failed to load SAML metadata: 500: try \d of 3/)
-        SAML::SettingsService.saml_settings(true)
+        SAML::SettingsService.merged_saml_settings(true)
+      end
+      it 'should keep making GET calls to fetch metadata' do
```

Comment:
[Stub out Kernel.sleep](http://stackoverflow.com/questions/1168845/rspec-stubbing-kernelsleep) to be a noop? Since you're interested in the number of calls and not the time spent sleeping it seems like that would be an ok approach.

Otherwise, if I'm reading it right you could remove the second call to `SAML::SettingsService.saml_settings`, cut it down to 6 tries and still be testing the same thing. I'm not sure how else to improve it.